src/Orleans.Core/Serialization/BinaryTokenStreamReader.cs

```diff
@@ -867,6 +865,14 @@ public byte ReadByte()
             return buff[offset];
         }
 
+        public byte PeekByte()
```

Comment:
should `PeekToken` reuse this ? I see the two methods are oddly alike.

Reply:
LGTM except some minor comments

Reply:
I think `PeekToken` does use this - see line 77 above

Reply:
Maybe I should delete the `BinaryTokenStreamReader.PeekToken` impl. We can do that in another PR, though